modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/index.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {render} from '@liferay/frontend-js-react-web';
+import React from 'react';
+
+import TranslationModal from './translation/TranslationModal';
+import TranslationSelector from './translation/TranslationSelector';
+
+export {TranslationModal};
+
+export default function (container, props) {
+	render(<TranslationSelector {...props} />, {}, container);
+}
```

Comment:
I'm guessing you just copied this over from the original module so it may have just been an oversight, but it is odd to me that we are adding a main entry point to `frontend-taglib-clay` and exporting a function to render a very specific component. This would make sense if it was its own module, but I don't think we should do this in a large "clay" module like this one.

Do you know where this component will be used at specifically or how it is used (in a react component or taglib)? Will it be used across multiple modules and in different UIs? I think this all goes back to the "where does this component live?" question. I read up on the LPS and I can't quite put together where this is going to be implemented at. If we know where this will be implemented at, it will help give an idea of how to expose it.

Reply:
Yes, you guessed right.

It is totally odd to be exporting one function that just renders a specific component - but I just copied the file from the `frontend-js-translation-web` module we decided to discard guessing we'd have a discussion about this.

The `TranslationSelector` and `TranslationModal` components could be used from any OSGi module that uses React ... but we also want to replace [this part](https://github.com/liferay/liferay-portal/blob/167f47e46c2ea38101a1cfbf6d2ba237e138b484/portal-web/docroot/html/taglib/ui/input_localized/page.jsp#L136-L222) of the  `InputLocalized` tag to render the `TranslationSelector` depending on an attribute (I haven't sent that yet) 

Maybe what I could do is export a "named" function instead of a default one that makes it more clear: for example `renderTranslationSelector`.

Reply:
> Maybe what I could do is export a "named" function instead of a default one that makes it more clear: for example renderTranslationSelector.

Yeah I like this idea, but instead of exporting a function that renders it, could we just export the component itself? That way we can defer the rendering bit to whoever is consuming it. That way it can be used both in existing React code or anywhere else.

This file would then just be

```js
import TranslationModal from './translation/TranslationModal';
import TranslationSelector from './translation/TranslationSelector';

export {TranslationSelector, TranslationModal};

// I don't know if it will complain about no default export... If it does, you can add an empty object for now
export default {}
```

Reply:
@bryceosterhaus that would be ideal. 
However in this case, I haven't found a way to define a function that renders the `TranslationSelector` in the `portal-web` module, so I defined a `renderTranslationSelector` function that I also export ... 

I'm thinking that maybe this is not the best solution: maybe the tag could load the required (JS) modules to make this happen, or maybe rendering the react component in that taglib doesn't make sense at all (after all, if this needs to be backported we could face issues). I'll speak with @dsanz about the issue just to see if he has ideas.